conda/core/link.py

```diff
@@ -524,14 +524,39 @@ def _verify(cls, prefix_setups, prefix_action_groups):
         return exceptions
 
     @classmethod
-    def _execute(cls, all_action_groups):
+    def _execute(cls, prefix_action_groups):
+        groups = list(interleave(itervalues(prefix_action_groups)))
```

Comment:
Can a tuple be used here rather than a list?  Perhaps creating a dictionary with action types would be even better so the actions do not need to be referred to by indices below.